### ISSUE TITLE
Issue Reporter

### ISSUE THREAD

**Issue by alexanderson1993** [Thorium-Sim/thorium-nova#83]
This is a proposal for an issue reporter feature. Thorium Classic had one which posted issues directly to Github, which was really nice.

<img width="518" alt="Screen Shot 2021-11-03 at 8 41 29 PM" src="https://user-images.githubusercontent.com/6558157/140238478-e05ff793-875e-4a92-844a-faa8dd258168.png">

A few things I don't like about it:

- These issues sometimes require discussions with the issue writer, but if they only provide their name it makes it hard to follow up. Email is better, but then the question becomes "Who will do the following up?" It's much better if the whole conversation can happen in Github.
- The body field supports markdown. It'd be great if we had some kind of rich text options, much like the Github comment field.
- Priority is a dumb field. Frankly, I don't care what priority the issue submitter thinks it is. The developers working on Thorium Nova should decide the priority of things.
- Attachment uploads never worked. It should be possible to upload multiple attachments.

**Describe the solution you'd like**

- For handling discussions, let's literally move it to Github. This adds some constraints: In order to submit an issue report in-app, the user must first sign in to Github. Which means they need a Github account. Not ideal, but it's the default I'm going to go with when I eventually implement this. **I would love other suggestions for how to address this.**
- The body field can use the same input field that I built for the admin pages on Thoriumsim.com. It's really nice, you'll like it, I promise.
- Remove the priority field. Easy peasy.
- Attachments can also work the same way as attachments for the Thoriumsim.com input field. I think it only supports images, but we can look into supporting other file types later.

As for implementation, I think it makes sense for the issue form to be tied directly to Thoriumsim.com. So when the form is submitted, it sends it to Thoriumsim.com, which then sends it to Github.com.

### Alternative Proposal

Instead of sending the issue directly to Github, we build out an issue tracker directly in Thoriumsim.com which links back to Github. I actually started building something like this last year. 

This might address the discussion problem, since folks could be required to sign in with their Thoriumsim.com account when submitting issues, which makes way more sense than signing in with a Github account. Discussion happens on Thoriumsim.com, and then once the issue is understood well enough, a Github issue can be created and triaged.

**Comment by Xansta** [comment]
An issue submission mechanism inside Thorium Nova is an excellent idea. It should hide the complexities of GitHub. 

I suggest making a contact method required. It could be email, it could be Discord name. By "required" I don't necessarily mean enforced on the submission form. Perhaps a note stating that including a contact method significantly improves the chance of action being taken on the issue reported. 

Yes, remove priority from the submission form. Priority can still be set in GitHub by the contribution team, right?

Please define OP
Owner/Proprietor?
Original Poster? <-- I assume something like this

Attachments would be nice, but anything is better than nothing

**Comment by alexanderson1993** [comment]
> I suggest making a contact method required. It could be email, it could be Discord name. 

In the issue, I mentioned that

> Email is better, but then the question becomes "Who will do the following up?" 

Having them share their email or Discord means that someone has to get in touch with them to get the follow-up information. I want this to be a community project, where everyone and anyone can take care of tasks like following up. So who does it? How do we decide who does it? and how do we make sure that follow-up information is made available to everyone else so they can help with addressing the issue?

Not an insurmountable problem, but certainly one that requires a bit of coordination on the part of the community. I'd much rather force issue writers to sign in with a Github account and take advantage of the great notification and commenting features that Github already provides. Bonus: By encouraging them to sign up, they are one step closer to becoming more involved contributors. 😉

> Priority can still be set in GitHub by the contribution team, right?

Yep, anyone with contribution access to the repo can change the priority setting on an issue. (If anyone doesn't have contribution access, file an issue or ask on Discord and we'll send you an invite to the Thorium-Sim org.)

OP = Original Poster. The person who authored the issue. I've updated the description to remove the "OP" term.

**Comment by alexanderson1993** [comment]
> since folks could be required to sign in with their Thoriumsim.com account when submitting issues,

I actually think this is the ideal approach, for a number of reasons:

- Folks will hopefully be signing into their Thorium account anyway for tracking their flights, downloading and submitting plugins. See more about what that account does here: https://github.com/Thorium-Sim/thorium-nova/discussions/21#user-content-thorium-accounts
- We can make it so they can create a Thorium account from within Thorium Nova.
- Make it possible to connect a Thorium account to Github - ideally this also can happen within Thorium Nova.
- Only allow folks to submit issues if they've connected their Github account.
- No discussion or storing the issues on ThoriumSim.com - it just proxies the submitted issue to Github.

Anyone have any thoughts about that?
